src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -81,28 +80,26 @@ object DepositHandler {
 
   def finalizeDeposit(mimeType: String)(implicit settings: Settings, id: String): Try[Unit] = {
     log.info(s"[$id] Finalizing deposit")
-    // TODO pass on the combination object of baseDir and baseURL
-    implicit val baseDir: File = new File(settings.bagStoreBaseDir)
-    implicit val baseUrl: URI  = new URI(settings.bagStoreBaseUri)
+    implicit val optBagStoreSettings: Option[BagStoreSettings] = settings.bagStoreSettings
     val tempDir = new File(settings.tempDir, id)
 
     val result = for {
-      _        <- checkBagStoreBaseDir()
+      _        <- optBagStoreSettings.map(checkBagStoreBaseDir).getOrElse(Success(()))
```

Comment:
Now the existence of the directory is checked once for each deposit.  If that is too much, then we have to 'build' something to make it happen less frequently.